Sources/xcodes/main.swift

```diff
@@ -220,17 +232,60 @@ struct Xcodes: ParsableCommand {
             }
             
             let destination = getDirectory(possibleDirectory: directory)
-            
-            installer.install(installation, dataSource: globalDataSource.dataSource, downloader: downloader, destination: destination, experimentalUnxip: experimentalUnxip)
-                .done { Install.exit() }
+
+            if select == false {
+                // install normally
+                installer.install(installation, dataSource: globalDataSource.dataSource, downloader: downloader, destination: destination, experimentalUnxip: experimentalUnxip)
+                    .done { Install.exit() }
+                    .catch { error in
+                        Install.processDownloadOrInstall(error: error)
+                    }
+            } else {
+                // check if the version is already installed and try to select it
+                firstly { () -> Promise<Void> in
+                    if case .version(let version) = installation {
+                        return selectXcode(shouldPrint: print, pathOrVersion: version, directory: destination, fallbackToInteractive: false)
+                    } else {
+                        return Promise { _ in
+                            throw InstallError.notAlreadyInstalled
```

Comment:
I'm not a huge fan of throwing just to get to a catch. Can you refactor this a bit so that it's a bit simpler around the different options. That may involved some new functions to make things a bit cleaner. 

Let me know if you need some help around this. Thanks

Reply:
I did some refactor. Let me know what you think.

Reply:
I think unit testing is where I'm kind of lost and will require some help